go/updater/updater.go

```diff
@@ -310,8 +315,18 @@ func (u *Updater) apply(src string, dest string) (tmpPath string, err error) {
 
 // Update checks, downloads and performs an update.
 func (u *Updater) Update(ui libkb.UpdateUI, force bool, requested bool) (update *keybase1.Update, err error) {
+	// If already running, return error
 	u.Lock()
-	defer u.Unlock()
+	if u.running {
+		err = fmt.Errorf("Already running update")
+		u.Unlock()
+		return
+	}
+	u.running = true
+	u.Unlock()
+	defer func() {
+		u.running = false
```

Comment:
Hmm... but you want it to fail if you can't grab the lock.. Hmm... we might have to make a little lock-derived class that has a Lock and Unlock but Lock fails if already locked.